### PR TITLE
{Monitor} Fix monitor test 7/7

### PR DESCRIPTION
**Pull request by haroldrandom**
**Description**

Against the latest live test report https://clitestresultstac.blob.core.windows.net/20201211-043721-979918/monitor.report.parallel.html

1. test `test_metric_alert_for_rg_and_sub` is untouched, just pass after rerun. No idea why live test failed.
2. test `test_monitor_clone_vm_metric_alerts_scenario` is untouched, just pass after rerun. No idea why live test failed.
3. test `test_monitor_log_analytics_workspace_data_export` is untouched, just pass after rerun. No idea why live test failed.
4. test `test_monitor_log_analytics_workspace_data_export` failed because service introduce a breaking change without notifying us and it's awarded by the user #15742. @kairu-ms will take care of it.
5. test `test_monitor_private_link_scope_scenario` is buggy due to service problem who complaining that can't find a just-created resource, so _record_only_ it.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_log_analytics_workspace.py [hunk 1]
```diff
@@ -461,7 +461,7 @@ def test_monitor_log_analytics_workspace_data_export(self, resource_group, accou
 
         self.cmd(
             'monitor log-analytics workspace data-export create -g {rg} --workspace-name {workspace_name} -n {data_export_name} '
-            '--destination {eventhub_id} --all --enable false',
+            '--destination {eventhub_id} --enable false -t {table_name}',
```
Comment (haroldrandom) [inline src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_log_analytics_workspace.py:464]: This modify is related to a breaking change, described in https://github.com/Azure/azure-cli/issues/15742

File: src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_private_link_scope.py [hunk 2]
```diff
@@ -51,6 +52,7 @@ def test_monitor_private_link_scope_scenario(self, resource_group, resource_grou
             'workspace_id': workspace_id
         })
 
+        # this command failed as service cannot find component of application insights
         self.cmd('monitor private-link-scope scoped-resource create -g {rg} -n {assigned_app} --linked-resource {app_id} --scope-name {scope}', checks=[
```
Comment (haroldrandom) [inline src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_private_link_scope.py:56]: Service is always complaining that cannot find a just-created app-insight component. So have to _record_only_ it

```
        if response.status_code not in [200, 201, 202]:
            exp = CloudError(response)
            exp.request_id = response.headers.get('x-ms-request-id')
>           raise exp
E           msrestazure.azure_exceptions.CloudError: Azure Error: PreconditionFailed
E           Message: Resource '/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/clitest.rgfpfgcxn6lts354fiisn66us22nbdcbjihethfyig26yfqbqtg5fmwxxb2jnv224qc/providers/microsoft.insights/components/clitestwqt2popxlabbl' not found
E           Inner error: {
E               "trace": [
E                   "Microsoft.AppInsights.Nexus.ResourceStore.ResourceStoreException"
E               ]
E           }
```

File: src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_private_link_scope.py [hunk 3]
```diff
@@ -12,7 +12,8 @@ def __init__(self, method_name, config_file=None, recording_dir=None, recording_
         super(TestMonitorPrivateLinkScope, self).__init__(method_name)
         self.cmd('extension add -n application-insights')
 
-    @ResourceGroupPreparer(location='eastus2euap')
+    @record_only()  # record_only as the private-link-scope scoped-resource cannot find the components of application insights
+    @ResourceGroupPreparer(location='westus2')
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_private_link_scope.py:16]: Why not change location back to 'eastus2euap'?

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_private_link_scope.py (hunk 3 above)
Comment (haroldrandom) [reply src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_private_link_scope.py:16]: `eastus2euap` is the wrong location for this service, we should correct it to let the true error expose.

**Comment by yonzhan** [comment]
Monitor

**Pull request merged**